hibernate-core/src/main/java/org/hibernate/mapping/Index.java

```diff
@@ -223,7 +224,22 @@ public String getName() {
 	}
 
 	public void setName(String name) {
-		this.name = name;
+		if (
+				StringHelper.isNotEmpty( name ) &&
+						Dialect.QUOTE.indexOf( name.charAt( 0 ) ) > -1 //TODO: deprecated, remove eventually
```

Comment:
As indicated below (as well as Christian's PR) really the proper solution imo is to track the Identifier here, not the String.